### PR TITLE
Tweak CI: Switch to Micromamba, pin to supported OpenFF Toolkit versions

### PR DESCRIPTION
**Pull request by mattwthompson**
This  might already be in a branch somewhere, but it should be easier to get through if it's a more atomic change. I noticed when working on something else that
* It takes CI ~10 minutes to set up a conda environment
* There is no `.gitignore` file in this repo

### CODE REVIEW

**Comment by mikemhenry** [comment]
I will review this soon, I will also fix the required checks that don't run anymore

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -10,6 +10,10 @@ on:
   schedule:
     - cron: "0 0 * * *"
 
+defaults:
```
Comment (mikemhenry) [inline .github/workflows/ci.yaml:13]: Thanks for doing this, it bugs me every time I touch the CI file (and others)

**Review by mikemhenry — APPROVED** [review]

**Comment by mikemhenry** [comment]
> Drop Python 3.7 
Let's just do that now

**Comment by mattwthompson** [comment]
Thanks!

**Comment by mikemhenry** [comment]
Thanks so much @mattwthompson for picking up my slack on maintenance of this repository.

**Pull request merged**